### ISSUE TITLE
Why can't pickle meterGroup？

### ISSUE THREAD

**Issue by HardyChina** [nilmtk/nilmtk#834]
I want to use multiprocessing to accelerate the computation,but the error is happen:
Can't pickle local object 'WeakValueDictionary.__init__.<locals>.remove'.
This error can be reproduced as follows:
pickle.dumps(test_elec).
I am a new hand for Pyhton,I not find WeakValueDictionary in meterGroup.

**Comment by HardyChina** [comment]
I find the question，datastore causes the problem。

**Comment by PMeira** [comment]
> I find the question，datastore causes the problem。

Yep, that's the issue. You probably need to wrap the offending code to replace the dataset reference with something picklable and restore it on the other side. If you look into some of the `export_model` functions, they do it to save the model to disk when it requires a reference to a dataset.

It might be worth to investigate an alternative built into NILMTK later, so let's keep this issue open for the time being.
